src/amr/data/particles/refine/split_2d.h

```diff
@@ -0,0 +1,481 @@
+                      Deltas& deltas, float const& delta_val)
+    {
+        Dispatcher::set_weights(weights, weight_val);
+        Dispatcher::set_deltas(deltas, delta_val);
+    }
+
+protected:
+    constexpr static float delta_val_  = 0.571783;
+    constexpr static float weight_val_ = 0.25;
+};
+
+
+template</*default pattern*/>
+class Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<4>>
+    : public Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<4>, Color::PURPLE>
+{
+};
+
+
+/**************************************************************************/
+template<>
+class Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<8>, Color::PINK, Color::PURPLE>
+    : public ASplitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<8>>
+{
+    using Dispatcher
+        = SplitParticleDispatcher<DimConst<2>, RefinedParticlesConst<8>, Shape::Square>;
+
+public:
+    Splitter() { split(weights_, deltas_); }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, Deltas& deltas)
+    {
+        split(weights, weight_vals_, deltas, delta_vals_);
+    }
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, float const weight_vals[], //
+                      Deltas& deltas, float const delta_vals[])
+    {
+        Dispatcher::set_weights(weights, weight_vals);
+        Dispatcher::set_deltas(deltas, delta_vals);
+    }
+
+protected:
+    constexpr static float delta_vals_[]  = {0.700909, 1.05786};
+    constexpr static float weight_vals_[] = {0.179488, 0.070512};
+};
+
+
+template</*default pattern*/>
+class Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<8>>
+    : public Splitter<DimConst<2>, InterpConst<1>, RefinedParticlesConst<8>, Color::PINK,
+                      Color::PURPLE>
+{
+};
+
+
+/**************************************************************************/
+template<>
+class Splitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<16>>
+    : public ASplitter<DimConst<2>, InterpConst<2>, RefinedParticlesConst<16>>
+{
+    using Dispatcher              = ExactSplitParticleDispatcher<DimConst<2>>;
+    static constexpr uint8_t root = 4;
+
+public:
+    Splitter() { split(weights_, deltas_); }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, Deltas& deltas)
+    {
+        split(weights, weight_vals_, deltas, delta_vals_);
+    }
+
+    template<typename Weights, typename Deltas>
+    static void split(Weights& weights, float const weight_vals[], //
+                      Deltas& deltas, float const delta_vals[])
+    {
+        for (size_t i : {0, 3, 12, 15}) // first loop = top left corner
+            weights[i] = weight_vals[ColorIndex::RED];
+
+        for (size_t i : {1, 2, 4, 7, 8, 11, 13, 14})
+            weights[i] = weight_vals[ColorIndex::ORANGE];
+
+        for (size_t i : {5, 6, 9, 10})
+            weights[i] = weight_vals[ColorIndex::GREEN];
```

Comment:
this one is `PURPLE`

Reply:
We have two indices that are purple here.

how about 

PURPLE_OUTER
PURPLE_INNER
?

Reply:
I'm not that sure we want to make a distinction between them... they are exactly the same, except their weights and delta value. is this needed in the code ?

Reply:
> except their weights and delta value. is this needed in the code

this is for distinguishing between the different weights/deltas for the same color

Reply:
the following doesn't compile 
![image](https://user-images.githubusercontent.com/30491/82902505-d60b5d80-9f5f-11ea-9183-23beb455b970.png)